src/hyperlight_host/src/mem/mgr.rs

```diff
@@ -80,6 +75,62 @@ pub(crate) struct SandboxMemoryManager<S> {
     pub(crate) abort_buffer: Vec<u8>,
 }
 
+#[cfg(feature = "init-paging")]
+struct GuestPageTableBuffer {
+    buffer: std::cell::RefCell<Vec<[PageTableEntry; PAGE_TABLE_ENTRIES_PER_TABLE]>>,
+}
+#[cfg(feature = "init-paging")]
+impl vm::TableOps for GuestPageTableBuffer {
+    type TableAddr = (usize, usize);
+    unsafe fn alloc_table(&self) -> (usize, usize) {
+        let mut b = self.buffer.borrow_mut();
+        let page_addr = b.len();
+        b.push([0; PAGE_TABLE_ENTRIES_PER_TABLE]);
+        (page_addr, 0)
+    }
+    fn entry_addr(addr: (usize, usize), offset: u64) -> (usize, usize) {
+        (addr.0, offset as usize >> 3)
```

Comment:
I'm not sure I understand what's going on here.
I thought it would be something like:
```
let resolved_addr = PAGE_TABLE_ENTRIES_PER_TABLE * addr.0 + addr.1 + offset;
(resolved_addr / PAGE_TABLE_ENTRIES_PER_TABLE, resolved_addr % PAGE_TABLE_ENTRIES_PER_TABLE)
```
or in other words
```
let phys = Self::to_phys(addr) + offset;
Self::from_phys(phys)
```
but that's not the case, so I'm definitely missing something.

Reply:
I don't think we should panic , I think we should either leave it as is or we should alter the parameters to make it correct